File: java/strings/palindrome_check.java
import java.util.Scanner;

public class PalindromeCheck {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a string: ");
        String str = sc.nextLine();

        // Convert string to char array
        char[] arr = str.toCharArray();

        int i = 0, j = arr.length - 1;
        boolean isPalindrome = true;

        while (i < j) {
            if (arr[i] != arr[j]) {
                isPalindrome = false;
                break;
            }
            i++;
            j--;
        }

        if (isPalindrome) {
            System.out.println("The string is a palindrome.");
        } else {
            System.out.println("The string is NOT a palindrome.");
        }
    }
}

File: python/strings/palindrome_check.py
s = input("Enter a string: ")

arr = list(s)
i, j = 0, len(arr) - 1
is_palindrome = True

while i < j:
    if arr[i] != arr[j]:
        is_palindrome = False
        break
    i += 1
    j -= 1

if is_palindrome:
    print("The string is a palindrome.")
else:
    print("The string is NOT a palindrome.")
